docs/src/app/(public)/(content)/react/utils/use-render/demos/render-callback/css-modules/index.tsx

```diff
@@ -0,0 +1,55 @@
+'use client';
+import * as React from 'react';
+import { useRender } from '@base-ui-components/react/use-render';
+import { mergeProps } from '@base-ui-components/react/merge-props';
+import styles from './index.module.css';
+
+type CounterState = {
+  odd: boolean;
+};
+
+interface CounterProps extends useRender.ElementProps<CounterState> {
+  className?: string;
+  onClick?: (event: React.MouseEvent) => void;
+  ['aria-label']?: string;
+}
+
+function Counter(props: CounterProps) {
+  const { render = <button />, ...otherProps } = props;
+  const [count, setCount] = React.useState(0);
+  const odd = count % 2 === 1;
+  const state = React.useMemo(() => ({ odd }), [odd]);
+
+  const defaultProps: mergeProps.Props<'button'> = {
```

Comment:
I suppose it would be good to explain the types of `useRender.ElementProps` and `mergeProps.Props` and how they're different